src/shared/pages/welcomePage/script.js

```diff
@@ -0,0 +1,238 @@
+      "[data-locale='welcomePageManageShortcuts']"
+    );
+    manageShortcutsText.remove();
+  }
+}
+
+/**
+ * Check the hotkeys for the browser launch and let the user know if
+ * they are not set.
+ *
+ * Also add a link to the chromium shortcuts page.
+ */
+export async function checkChromiumHotkeys() {
+  const hotkeys = await browser.commands.getAll();
+  const launchBrowser = hotkeys.find(
+    (hotkey) => hotkey.name === "launchBrowser"
+  );
+  const launchFirefoxPrivate = hotkeys.find(
+    (hotkey) => hotkey.name === "launchFirefoxPrivate"
+  );
+
+  const shortcutsList = document.getElementById("shortcuts-list");
+
+  // if there are no shortcuts, display a message
+  if (!launchBrowser.shortcut && !launchFirefoxPrivate.shortcut) {
+    const preamble = document.createElement("p");
+    shortcutsList.appendChild(preamble);
+    replaceMessage(
+      preamble,
+      "welcomePageNoShortcutsChromium",
+      "chrome://extensions/shortcuts"
+    );
+
+    // remove the manage shortcuts text
+    const manageShortcutsText = document.querySelector(
+      "[data-locale='welcomePageManageShortcuts']"
+    );
+    manageShortcutsText.remove();
+    return;
+  }
+
+  // Display a message for the Firefox shortcut
+  const span = document.createElement("span");
+  span.classList.add("shortcut-item");
+  if (launchBrowser.shortcut) {
+    const launchBrowserHotkey = launchBrowser.shortcut;
+    span.innerText =
+      browser.i18n.getMessage("welcomePageYesShortcutTo", [
+        launchBrowserHotkey.toUpperCase(),
+        await getExternalBrowser(),
+      ]) + "\n";
+  } else {
+    span.innerText =
+      browser.i18n.getMessage("welcomePageNoShortcutTo", ["Firefox"]) + "\n";
+  }
+  shortcutsList.appendChild(span);
+
+  // Display a message for the Firefox private shortcut
+  const span2 = document.createElement("span");
+  span2.classList.add("shortcut-item");
+  if (launchFirefoxPrivate.shortcut) {
+    const launchFirefoxPrivateHotkey = launchFirefoxPrivate.shortcut;
+    span2.innerText =
+      browser.i18n.getMessage("welcomePageYesShortcutTo", [
+        launchFirefoxPrivateHotkey.toUpperCase(),
+        "Firefox private browsing",
+      ]) + "\n";
+  } else {
+    span2.innerText =
+      browser.i18n.getMessage("welcomePageNoShortcutTo", [
+        "Firefox private browsing",
+      ]) + "\n";
+  }
+  shortcutsList.appendChild(span2);
+}
+
+/**
+ * Add or remove certain elements from the page depending on the platform.
+ */
+export async function activatePlatformSpecificElements() {
+  if (isChromium) {
+    // remove objects with class firefox from the page
+    const firefoxElements = document.getElementsByClassName("firefox");
+    const firefoxElementsArray = Array.from(firefoxElements);
+    firefoxElementsArray.forEach((element) => {
+      element.remove();
+    });
+    checkChromiumHotkeys();
+    checkPrivateBrowsing();
+    if (browser.runtime.getPlatformInfo().os === "android") {
```

Comment:
Interesting... Is this extension expected to be run on a Chromium fork that supports extensions on Android? And if yes, is launching Firefox from it supported?

FYI: Google Chrome on Android does not support Chrome extensions, but there is an infrequently updated Chromium fork that does support Chrome extensions (infrequently updated = insecure).

Reply:
In the UI spec, a mobile layout was given, so I had assumed support. [see here](https://www.figma.com/file/7lgTI6V7gu3aLqjVhD2qVH/Dual-Browser-Extension?type=design&node-id=210-1225&mode=design&t=9LeqHlUn3FBEZlNv-0). But there seems to be no mention of this in the actual spec, so i doubt that it is truly a requirement for now. I will bring it up